components/ephemeral_storage/ephemeral_storage_service.cc

```diff
@@ -327,6 +327,13 @@ void EphemeralStorageService::ScheduleFirstPartyStorageAreasCleanupOnStartup() {
 
 void EphemeralStorageService::CleanupFirstPartyStorageAreasOnStartup() {
   DCHECK(!context_->IsOffTheRecord());
+
+  Profile* profile = Profile::FromBrowserContext(context_);
+  if (profile->HasAnyOffTheRecordProfile()) {
```

Comment:
maybe the better approach here is to detect if any "normal" profile windows are opened or not.

Reply:
Thanks for suggestion, I will try to follow it.

Reply:
Done. Please have a look